src/api/app/components/notification_avatars_component.rb

```diff
@@ -13,7 +13,8 @@ def avatar_objects
     @avatar_objects ||= if @notification.notifiable_type == 'Comment'
                           commenters
                         else
-                          @notification.notifiable.reviews.in_state_new.map(&:reviewed_by) + User.where(login: @notification.notifiable.creator)
+                          reviews = @notification.notifiable.reviews
+                          reviews.select(&:new?).map(&:reviewed_by) + User.where(login: @notification.notifiable.creator)
```

Comment:
Besides bullet is used the Prosopite Gem to identify n+1 queries, this is the only area where I still get complains `User.where(login: @notification.notifiable.creator)`

Reply:
```
N+1 queries detected:
  SELECT `users`.* FROM `users` WHERE `users`.`login` = 'Requestor'
  SELECT `users`.* FROM `users` WHERE `users`.`login` = 'Admin'
Call stack:
  /obs/src/api/app/components/notification_avatars_component.rb:17:in `avatar_objects'
  /obs/src/api/app/components/notification_avatars_component.rb:26:in `number_of_hidden_users'
  /obs/src/api/app/components/notification_avatars_component.html.haml:2:in `call'
  /usr/lib64/ruby/2.5.0/monitor.rb:235:in `mon_synchronize'
  /obs/src/api/app/components/notification_component.html.haml:25:in `call'
  /usr/lib64/ruby/2.5.0/monitor.rb:235:in `mon_synchronize'
  /obs/src/api/app/views/webui/users/notifications/_notifications_list.html.haml:24:in `block in _app_views_webui_users_notifications__notifications_list_html_haml__2629228725245621329_69931641932060'
  /obs/src/api/app/views/webui/users/notifications/_notifications_list.html.haml:16:in `_app_views_webui_users_notifications__notifications_list_html_haml__2629228725245621329_69931641932060'
  /obs/src/api/app/views/webui/users/notifications/index.html.haml:16:in `_app_views_webui_users_notifications_index_html_haml__4293752019784072337_47145832178840'
  /usr/lib64/ruby/2.5.0/benchmark.rb:308:in `realtime'
  /obs/src/api/app/controllers/webui/webui_controller.rb:29:in `n_plus_one_detection'
  /obs/src/api/config/initializers/wrap_parameters.rb:37:in `call'
```